packages/database/src/entity/DAG.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  Entity,
+  PrimaryGeneratedColumn,
+  ManyToOne,
+  ManyToMany,
+  JoinTable,
+} from 'typeorm'
+import { container } from '../data-source'
+
+import { User, Degree, Module } from '.'
+import { ModuleRepository } from '../repository/Module'
+import { DAGRepository } from '../repository/DAG'
+
+@Entity({ name: 'DAG' })
+export class DAG {
+  @PrimaryGeneratedColumn('uuid')
+  id: number
+
+  @ManyToOne(() => User)
+  @JoinTable()
+  user: User
+
+  @ManyToOne(() => Degree)
+  @JoinTable()
+  degree: Degree
+
+  @ManyToMany(() => Module)
+  @JoinTable()
+  modulesPlaced: Module[]
+
+  @ManyToMany(() => Module)
+  @JoinTable()
+  modulesHidden: Module[]
+
+  /**
+   * Toggle a Module's status between placed and hidden.
+   * @param {string} moduleCode
+   */
+  async toggleModule(moduleCode: string): Promise<DAG> {
```

Comment:
can discuss if we should
- return a DAG
- return void (then devs need to do a `.find()` again to get the updated entity)

If we go for returning a DAG, then we should update `Degree.insertModules()`